File: js/1_twoSum.js
'use strict';

/*Given an array of integers nums and
an integer target, return indices of
the two numbers such that they add
up to target*/

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */



var twoSum = function(nums, target) {
    let numMap = new Map();

    for (let i = 0; i<nums.length; i++) {
        if (numMap.has(nums[i])) {
            return [numMap.get(nums[i]), i];
        };
        numMap.set(target - nums[i], i);
    };
};
    


// var twoSum = function(nums, target) {

//     let init = [...nums];
//     nums = nums.sort((a, b) => a-b);
//     let i = 0;
//     let j = nums.length-1;
//     while (true) {
//         let sum = nums[i] + nums[j];
//         if (sum === target) break;
//         if (sum > target) {
//             j--;
//         } else { 
//             i++;
//         };
//     };

//     let l = init.findIndex(e => e === nums[i]);
//     console.log(l)
//     for (let k = init.length-1; k >= 0; k--) {
//         console.log()
//         if (init[k] === nums[j]) {
//             console.log(k)
//             return [l, k] }
//     }
// };

File: golang/1_twoSum.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
    numsMap := make(map[int]uint16);
	for i := uint16(0); i < uint16(len(nums)); i++ {
		var diff int = target - nums[i];
		_, ok := numsMap[diff];
		if (!ok) {
			numsMap[nums[i]] = i;
		} else {
			return []int{int(numsMap[diff]), int(i)}
		}
	};
	return []int{};
};

func main() {
    sli := twoSum([]int{5, 25, 75}, 100);
	fmt.Println(sli)
}
